test/boost/memtable_test.cc

```diff
@@ -1170,3 +1173,166 @@ SEASTAR_TEST_CASE(flushing_rate_is_reduced_if_compaction_doesnt_keep_up) {
     });
 }
 
+static future<> exceptions_in_flush_helper(std::unique_ptr<sstables::file_io_extension> mep, bool& should_fail, const bool& did_fail) {
+    auto ext = std::make_shared<db::extensions>();
+    auto cfg = seastar::make_shared<db::config>(ext);
+
+    ext->add_sstable_file_io_extension("test", std::move(mep));
+
+    co_await do_with_cql_env([&](cql_test_env& env) -> future<> {
+
+        co_await env.execute_cql(fmt::format("create table t0 (pk text primary key, v text)"));
+
+        should_fail = true;
+
+        int i = 0;
+
+        BOOST_TEST_MESSAGE("Wait for fail");
+
+        auto f = make_ready_future<>();
+
+        while (!did_fail) {
+            std::string pk = "apa" + std::to_string(i++);
+            std::string v = "ko";
+            co_await env.execute_cql(fmt::format("insert into ks.t0 (pk, v) values ('{}', '{}')", pk, v));
+
+            f = f.then([&] {
+                return env.db().invoke_on_all([] (replica::database& db) {
+                    return db.flush_all_memtables();
+                });
+            });
+        }
+
+        BOOST_REQUIRE(did_fail);
+        BOOST_TEST_MESSAGE("Reset fail trigger");
+
+        should_fail = false;
+        bool isolated = false;
+        for (int i = 0; i < 10; ++i) {
+            isolated = co_await env.get_storage_service().invoke_on(0, [&](service::storage_service& ss) {
+                return ss.is_isolated(); 
+            });
+            if (isolated) {
+                break;
+            }
+            // isolation is not syncnronous;
+            co_await sleep(2s);
```

Comment:
We have the `eventually()` and `eventually_true()` helpers, which already implement something like this, see https://github.com/scylladb/scylladb/blob/master/test/lib/eventually.hh.

Reply:
Neat.

Reply:
Does not work here, because neither context in this case is a seastar thread